packages/react-sdk/src/components/components-utils.ts

```diff
@@ -0,0 +1,86 @@
+import * as components from "./components";
+import { registeredComponents } from "./index";
+
+export type ComponentName = keyof typeof components;
+
+/**
+ * We need to define component names manually here, instead of using Object.keys(components)
+ * Otherwise every component would be included in the bundle, even if it is not used
+ */
+const componentNames = [
+  "Box",
+  "Body",
+  "TextBlock",
+  "Heading",
+  "Paragraph",
+  "Link",
+  "RichTextLink",
+  "Span",
+  "Bold",
+  "Italic",
+  "Superscript",
+  "Subscript",
+  "Button",
+  "Input",
+  "Form",
+  "Image",
+  "Blockquote",
+  "List",
+  "ListItem",
+] as const;
```

Comment:
Instead of the `Equals` dance you could likely just do this:
```suggestion
] as const satisfies readonly [ComponentName, ...ComponentName[]];
```

Reply:
our prettier don’t like satisfies ;) will update

Reply:
and actually, at first I tried this without `as const` but thanks to `as const` you can even skip the tuple type complexity and simplify this like this:
```suggestion
] as const satisfies readonly ComponentName[];
```

Reply:
satisfies also doesn't give an expected result https://www.typescriptlang.org/play?#code/MYewdgzgLgBKC2AHcBTMUIwLwwN4CgYiYAhEADwC4YAKASiwD4wBXAGzYBpDiS2BDAEYDq9Jqw7cAvgG58+FOWQAnWFACeiFDADCIJKnQA5fvG04A1inUgAZjA1a7cfcjBoMc-KEiwEbjxMzTBwAbR4iACIyckjOYgSYfABdGH5MH2gYCH4oAEsIWzyUTGUUfgATcDZ1GFC9A3djUxR4gDoOhoDms1Dk5PkAekGYAEkwGHha4HTtRVNENm0hEAA3bXgWLIBzPPW0iZRlZRBleIBBABlLmCt1TE2swW08if9DKCCSwiA

Reply:
Ah, yes - you are right. Can't you just cast `Object.keys` here though? Derive the type + derive the runtime value, sure there is nothing stopping you from casting to a wrong value etc but the whole "unsafe" bit stays contained in a single, very simple, line so it's not that likely that this would end up being a problem.

Reply:
Object.keys immediately increase bundle size in a task I'm solving. All components becomes not tree-shakeable

Reply:
Then this is the only thing that comes to my mind right now that would avoid `Equals`:
https://www.typescriptlang.org/play?#code/MYewdgzgLgBKC2AHcBTMUIwLwwN4CgYiYAhEADwC4YAKASiwD4wBXAGzYBpDiS2BDAEYDq9Jqw7cAvgG58+FOWQAnWFACeiFDADCIJKnQA5fvG04A1inUgAZjA1a7cfcjBoMc-KEiwEbjxMzTBwAeUEAKxRgKAA6K3UIGh4iAmJ00gpqAEZuDOIAegLSAWF+HJSYKRgIfigASwhbepRMXBgAbQBpGHqwGATnRxRnf0MMAF0cqvw6GH5MAEFlZX51AB5B+2HR13GIRnkimABJfvh1OAXtRVNENm0hEAA3bXgWaBgAc3rX+f6UCsQMpODBFgAZcEDayYd6fQTaPouAzuYymVqEIA

I'm not super sold if it's that simpler than `Equals` though. It has a nice benefit of the error being more local though and more readable. With the current `Equal` strategy this is all you get:
> Argument of type 'true' is not assignable to parameter of type 'false'.(2345)
https://www.typescriptlang.org/play?#code/MYewdgzgLgBKC2AHcBTMUIwLwwN4CgYiYAhEADwC4YAKASiwD4wBXAGzYBpDiS2BDAEYDq9Jqw7cAvgG58+FOWQAnWFACeiFDADCIJKnQA5fvG04A1inUgAZjA1a7cfcjBoMc-KEiwEbjxMzTBwAbR4iACIyckjOGAiYAHokmGiBYX5I-ABdGH5MH2h5FJgASTAYeHU4Au1FU0Q2bSEQADdteBZoGABzAEsO-MqUZWUQZXiAQQAZGZgrdUwunsFtfsr-QygglAhCfEdtAFEARxZ+NggAHin4kkZsGFC70jzFKDQAE0xQkmm8gB+BzKFjaai2S4QFBeIpqTR7M4XNhPW73Rg0fjUJFQtGkRh0bCPFhgL4oWwbFBfORHCA4tjXPQGdzGUwoeJHZxbFk7NkQUKseBrZQ5DFQUEoOgyIA

The variant with `Object.keys` and a dummy object even comes with autocompletes for the missing keys etc

Reply:
I like your solution, Thank you!